pinot-common/src/main/java/org/apache/pinot/common/utils/config/QueryOptionsUtils.java

```diff
@@ -190,4 +191,25 @@ public static Integer getGroupTrimThreshold(Map<String, String> queryOptions) {
     String groupByTrimThreshold = queryOptions.get(QueryOptionKey.GROUP_TRIM_THRESHOLD);
     return groupByTrimThreshold != null ? Integer.parseInt(groupByTrimThreshold) : null;
   }
+
+  // Sample input: k1:1/k1:2/k1:3/k2:4/k2:5/k2:6
+  // Sample output: {k1=[1, 2, 3], k2=[4, 5, 6]}
+  public static Map<String, Set<Integer>> getColumnPartitionMap(Map<String, String> queryOptions) {
+    Map<String, Set<Integer>> columnPartitionMap = new HashMap<>();
+    String columnPartitionConfig = queryOptions.get(QueryOptionKey.COLUMN_PARTITION_MAP);
+    if (columnPartitionConfig != null) {
+      String[] columnPartitions = columnPartitionConfig.split("/");
```

Comment:
If we insist to support this syntax, I would include a preprocess phase where `\/` means to escape a `/`, `\:` means to skip a `:` and `\\` means to skip a `\`